style.md

```diff
@@ -154,7 +154,26 @@ An interface is two fields:
   the data stored is a value, then a pointer to the value is stored.
 
 If you want interface methods to modify the underlying data, you must use a
-pointer.
+pointer of object in assignment statement.
```

Comment:
This isn't exactly true: if the method receiver is a pointer but the type being used is an _addressable value_, it will be implicitly converted to a pointer for that call:

```go
type Foo struct {
  bar string
}

func (f *Foo) SetBar(value string) {
  f.bar = value
}

func (f *Foo) Bar() string {
  return f.bar
}

var f Foo
f.SetBar("hello")

fmt.Println(f.Bar())
// prints "hello"
```

You can see an example [here](https://play.golang.org/p/EX3EvSU0D2e).

Reply:
The example with property string and pointer maybe it could be changed as why would one use a pointer receiver on a primitive type?

Reply:
Choosing when to use a pointer or a value receiver is different than satisfying interfaces with pointer or value types (just so we don't conflate that with this PR).

That said, the official Go convention for using pointer or value receivers is pretty unambiguous: see "[Should I define methods on values or pointers?](https://golang.org/doc/faq#methods_on_values_or_pointers)" in the [Go FAQ](https://golang.org/doc/faq).